shared/src/commonMain/kotlin/org/hyperskill/app/share_streak/cache/ShareStreakCacheDataSourceImpl.kt

```diff
@@ -0,0 +1,17 @@
+package org.hyperskill.app.share_streak.cache
+
+import com.russhwolf.settings.Settings
+import org.hyperskill.app.share_streak.data.source.ShareStreakCacheDataSource
+
+internal class ShareStreakCacheDataSourceImpl(private val settings: Settings) : ShareStreakCacheDataSource {
+    override fun getLastTimeShareStreakShownEpochMilliseconds(): Long? =
+        settings.getLongOrNull(ShareStreakCacheKeyValues.SHARE_STREAK_LAST_TIME_SHOWN)
+
+    override fun setLastTimeShareStreakShown(epochMilliseconds: Long) {
+        settings.putLong(ShareStreakCacheKeyValues.SHARE_STREAK_LAST_TIME_SHOWN, epochMilliseconds)
+    }
+
+    override fun clearCache() {
+        settings.remove(ShareStreakCacheKeyValues.SHARE_STREAK_LAST_TIME_SHOWN)
+    }
```

Comment:
I think we can use `Long` only on level of data store. For me it's better to work with `Instant` instead of `Long`. Something like that:

```
override fun setLastTimeShareStreakShown(instant: Instant) {
        settings.putLong(ShareStreakCacheKeyValues.SHARE_STREAK_LAST_TIME_SHOWN, instant.toEpochMilliseconds())
    }
```